Formula/zsh-syntax-highlighting.rb

```diff
@@ -35,6 +35,6 @@ def caveats
 
   test do
     assert_match "#{version}\n",
-      shell_output("zsh -c '. #{share}/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh && echo $ZSH_HIGHLIGHT_VERSION'")
+      shell_output("zsh -c '. #{pkgshare}/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh && echo $ZSH_HIGHLIGHT_VERSION'")
```

Comment:
Should be: `#{pkgshare}/zsh-syntax-highlighting.zsh` as otherwise you've changed the path it's looking at.